mllib/src/main/scala/org/apache/spark/ml/feature/Word2Vec.scala

```diff
@@ -146,6 +151,55 @@ class Word2VecModel private[ml] (
     wordVectors: feature.Word2VecModel)
   extends Model[Word2VecModel] with Word2VecBase {
 
+
+  /**
+   * Return a map of every word to its Vector representation.
+   */
+  val getVectors: Map[String, Array[Float]] = wordVectors.getVectors
+
+  /**
+   * Java friendly version of getVectors called by the python wrappers.
+   */
+  private[spark] val javaGetVectors: JList[Object] = {
+    val (words, vectors) = getVectors.toSeq.unzip
+
+    // There are problems with serialization of Arrays which makes this conversion
+    // necessary.
+    val vectorsList = vectors.toList.map(vec => Vectors.dense(vec.map(_.toDouble)))
+    List(words.toList, vectorsList).map(_.asJava.asInstanceOf[Object]).asJava
+  }
+
+  /**
+   * Find "num" no. words closest to similarity to the given word.
+   * Returns a coupled array with the second element giving the similarity measure.
+   */
+  def findSynonyms(word: String, num: Int): Array[(String, Double)] = {
```

Comment:
Similar to the comment above, I don't think we can access the Spark Context here, although perhaps one option (and I don't really know if its a good one) would be to do something similar to the predict method and allow take either a dataframe or rdd of multiple inputs and return a dataframe/rdd of the synonyms.